crates/services/bls_aggregation/src/bls_agg.rs

```diff
@@ -31,6 +31,65 @@ pub struct AggregatedOperators {
     pub signers_operator_ids_set: HashMap<FixedBytes<32>, bool>,
 }
 
+#[derive(Clone)]
+pub struct TaskMetadata {
+    /// Index of the task
+    task_index: TaskIndex,
+    /// Block the task was created
+    task_created_block: u32, // TODO: Should this be a u64?
+    /// Quorum numbers which should respond to the task
+    quorum_numbers: Vec<u8>,
+    /// Thresholds for each quorum
+    quorum_threshold_percentages: QuorumThresholdPercentages,
+    /// Time before expiry of the task response aggregation
+    time_to_expiry: Duration,
+    // Duration of the window to wait for signatures after quorum is reached
+    window_duration: Duration,
+}
+
+impl TaskMetadata {
+    /// Creates a new instance of the TaskMetadata
+    ///
+    /// # Arguments
+    /// * `task_index` - The index of the task
+    /// * `task_created_block` - The block number at which the task was created
+    /// * `quorum_numbers` - The quorum numbers which should respond to the task
+    /// * `quorum_threshold_percentages` - The threshold percentages for each quorum
+    /// * `time_to_expiry` - The time before expiry of the task response aggregation
+    ///
+    /// Use the `with_window_duration` method to set the window duration.
```

Comment:
You can use links to add link to specific code symbols (the short syntax is also available)
```suggestion
    /// Use [`with_window_duration`](Self::with_window_duration) to set the window duration.
```